.github/workflows/pr-validation.yml

```diff
@@ -1,28 +1,58 @@
+# Code validation job, executed on pull request
+# Builds the app, and runs lint, unit and integration tests
+
 name: PR Validation
 
 on:
   pull_request:
-    branches: [ "main" ]
+    branches: [ main ]
+    paths-ignore:
+      - '.github/ISSUE_TEMPLATE/**'
+      - 'assets/locales/**'
+      - 'docs/**'
+      - 'sample-session-files/**'
+      - '.gitignore'
+      - 'LICENSE'
+      - 'README.md'
+      - 'renovate.json'
 
 jobs:
   build:
+    name: Build & Test
     strategy:
       matrix:
-        node-version: [16.x, 18.x]
         image: [ubuntu-latest, macos-latest, windows-latest]
-
     runs-on: ${{ matrix.image }}
 
     steps:
     - uses: actions/checkout@v3
     - name: Prepare Xvfb if needed
       if: ${{ matrix.image == 'ubuntu-latest' }}
       run: Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
-    - name: Use Node.js ${{ matrix.node-version }}
+    - name: Use Node.js 18.x
+      uses: actions/setup-node@v3
+      with:
+        node-version: 18.x
+        cache: 'npm'
+    - name: Install Dependencies
+      run: npm ci
+    - name: Build
+      run: npm run build --if-present
+    - name: Test
+      run: npm run test:unit && npm run test:integration
```

Comment:
do unit tests need xvfb?
I would also run them in parallel with integration to have better perf

Reply:
For now it looks like neither unit nor integration tests require a display. I'm guessing this was meant more for e2e tests.
I'm not sure there's much benefit to running the current test suites in parallel. On my mac mini the unit tests take 350ms, and integration tests take 4s. The main bulk of time in this workflow is spent on the previous build step - perhaps it's better to extract that step?

Reply:
After removing xvfb, the Ubuntu runner job still passes, so this removal appears to be safe.